akka-remote/src/main/scala/akka/remote/artery/SendQueue.scala

```diff
@@ -42,26 +43,37 @@ private[remote] final class SendQueue[T] extends GraphStageWithMaterializedValue
 
   override def createLogicAndMaterializedValue(inheritedAttributes: Attributes): (GraphStageLogic, QueueValue[T]) = {
     @volatile var needWakeup = false
-    @volatile var queue: Queue[T] = null
+    @volatile var producerQueue: Queue[T] = null
```

Comment:
that is the underlying queue, that is injected from the outside with the `inject` method of the materialized `QueueValue`.

there is one ordinary `var` for the consumer side inside the stage, that is set via the promise

in the latest I moved this `var` for the producer side to `QueueValue`. it still must be a `var` since it's injected